src/site/content/en/blog/webassembly-threads/index.md

```diff
@@ -0,0 +1,472 @@
+UI.
+
+In a simple application such as the previous example  `-s PROXY_TO_PTHREAD` is the best option:
+
+```shell
+emcc -pthread -s PROXY_TO_PTHREAD example.c -o example.js
+```
+
+### C++
+
+All the same caveats and logic apply in the same way to C++. The only new thing you gain is access
+to higher-level APIs like [`std::thread`](https://en.cppreference.com/w/cpp/thread/thread) and
+[`std::async`](https://en.cppreference.com/w/cpp/thread/async), which use the previously discussed
+`pthread` library under the hood.
+
+So the example above can be rewritten in more idiomatic C++ like this:
+
+```cpp/1,9,14
+#include <iostream>
+#include <thread>
+#include <chrono>
+
+int main()
+{
+    puts("Before the thread");
+
+    int arg = 42;
+    std::thread thread([&]() {
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+        std::cout << "Inside the thread: " << arg << std::endl;
+    });
+
+    thread.join();
+
+    std::cout << "After the thread" << std::endl;
+
+    return 0;
+}
+```
+
+When compiled and executed with similar parameters, it will behave in the same way as the C example:
+
+```shell
+emcc -std=c++11 -pthread -s PROXY_TO_PTHREAD example.cpp -o example.js
+```
+
+Output:
+
+```text
+Before the thread
+Inside the thread: 42
+Pthread 0xc06190 exited.
+After the thread
+Proxied main thread 0xa05c18 finished with return code 0. EXIT_RUNTIME=0 set, so keeping
+main thread alive for asynchronous event operations.
+Pthread 0xa05c18 exited.
+```
+
+### Rust
+
+Unlike Emscripten, Rust doesn't have a specialised end-to-end Web target, but instead provides a
+generic `wasm32-unknown-unknown` target for generic WebAssembly output.
+
+If Wasm is intended to be used in a Web environment, any interaction with JavaScript APIs is left to
+external libraries and tooling like [wasm-bindgen](https://rustwasm.github.io/docs/wasm-bindgen/)
+and [wasm-pack](https://rustwasm.github.io/docs/wasm-pack/). Unfortunately, this means that the
+standard library is not aware of Web Workers and standard APIs like
+[`std::thread`](https://doc.rust-lang.org/std/thread/) won't work when compiled to WebAssembly.
+
+Luckily, the majority of the ecosystem depends on higher-level libraries to take care of
+multithreading. At that level it's much easier to abstract away all the platform differences.
+
+In particular, [Rayon](https://crates.io/crates/rayon) is the most popular choice for
+data-parallelism in Rust. It allows to take method chains on regular iterators and, usually with a
+single line change, convert them in a way where they'd run in parallel on all available threads
+instead of sequentially. For example:
+
+```rust/3/2
+pub fn sum_of_squares(numbers: &[i32]) -> i32 {
+  numbers
+  .iter()
+  .par_iter()
+  .map(x => x * x)
+  .sum()
+}
+```
+
+This unassuming code will split up the input data, calculate `x * x` and partial sums in parallel
+threads, and in the end add up those partial results together.
```

Comment:
```suggestion
This code will split up the input data, calculate `x * x` and partial sums in parallel
threads, and in the end add up those partial results together.
```

Reply:
Unnecessary adjective.

Reply:
I somehow want to emphasise that the difference is really just a `par_` prefix that makes all of this to work, even though we're not manually creating and dealing with threads anywhere like we did in C/C++ examples.

Maybe rewording to "With this small change, the code ..." would be better?

Reply:
If the emphasis is important, then yes, that's better phrasing. Go for it 👍